test/data_generation.jl

```diff
@@ -1,18 +1,23 @@
 @testset "Data Generation" begin
+    function _is_well_conditioned(matrix::Array{Float64})
+        @test !isempty(matrix)
+        @test isposdef(matrix)
+        @test cond(matrix) < 20
```

Comment:
else the `20` number is truly magic 
```suggestion
        @asset length(matrix) ≤ 25  
        @test cond(matrix) < 20
```